src/OpenApi/test/Extensions/OpenApiEndpointRouteBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,155 @@
+
+        // Act
+        var requestDelegate = endpoint.RequestDelegate;
+        await requestDelegate(context);
+
+        // Assert
+        Assert.Equal(StatusCodes.Status200OK, context.Response.StatusCode);
+        ValidateOpenApiDocument(responseBodyStream, document =>
+        {
+            Assert.Equal("OpenApiEndpointRouteBuilderExtensionsTests", document.Info.Title);
+            Assert.Equal("1.0.0", document.Info.Version);
+        });
+    }
+
+    [Fact]
+    public async Task MapOpenApi_ReturnsDefaultDocumentIfNoNameProvided()
+    {
+        // Arrange
+        var hostEnvironment = new HostEnvironment() { ApplicationName = nameof(OpenApiEndpointRouteBuilderExtensionsTests) };
+        var serviceProviderIsService = new ServiceProviderIsService();
+        var serviceProvider = new ServiceCollection()
+            .AddSingleton<IServiceProviderIsService>(serviceProviderIsService)
+            .AddSingleton<IHostEnvironment>(hostEnvironment)
+            .AddOpenApi()
+            .BuildServiceProvider();
+        var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(serviceProvider));
+        builder.MapOpenApi("/openapi.json");
+        var context = new DefaultHttpContext();
+        var responseBodyStream = new MemoryStream();
+        context.Response.Body = responseBodyStream;
+        context.RequestServices = serviceProvider;
+        var endpoint = builder.DataSources.First().Endpoints.First();
+
+        // Act
+        var requestDelegate = endpoint.RequestDelegate;
+        await requestDelegate(context);
+
+        // Assert
+        Assert.Equal(StatusCodes.Status200OK, context.Response.StatusCode);
+        ValidateOpenApiDocument(responseBodyStream, document =>
+        {
+            Assert.Equal("OpenApiEndpointRouteBuilderExtensionsTests", document.Info.Title);
+            Assert.Equal("1.0.0", document.Info.Version);
+        });
+    }
+
+    [Fact]
+    public async Task MapOpenApi_Returns404ForUnresolvedDocument()
+    {
+        // Arrange
+        var hostEnvironment = new HostEnvironment() { ApplicationName = nameof(OpenApiEndpointRouteBuilderExtensionsTests) };
+        var serviceProviderIsService = new ServiceProviderIsService();
+        var serviceProvider = new ServiceCollection()
+            .AddSingleton<IServiceProviderIsService>(serviceProviderIsService)
+            .AddSingleton<IHostEnvironment>(hostEnvironment)
+            .AddOpenApi()
+            .BuildServiceProvider();
+        var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(serviceProvider));
+        builder.MapOpenApi();
+        var context = new DefaultHttpContext();
+        var responseBodyStream = new MemoryStream();
+        context.Response.Body = responseBodyStream;
+        context.RequestServices = serviceProvider;
+        context.Request.RouteValues.Add("documentName", "v2");
+        var endpoint = builder.DataSources.First().Endpoints.First();
```

Comment:
It might be easier in the long run to convert these to use TestServer.

Reply:
I don't expect us to add too many additional tests to this particular codepath after this PR. I have an alternate strategy for validating the contents of the OpenAPI document that actually gets generated.